### ISSUE TITLE
Labels for issues

### ISSUE THREAD

**Issue by AEA2002** [FreeCodeCamp/FreeCodeCamp#3371]
Is there information on the meaning for the various labels used for the issues. I have been searching and haven't found anything. Some of the labels are not clear (to me):

Does *test iimprovement* mean that the person who submitted the issue should test the change or anyone can test it or...? Should members wait for a *help wanted* label before attempting a fix for an issue?

Maybe something info about the labels could be added to the wiki?

Thanks.

**Comment by bugron** [comment]
**Account Problem** - camper has a problem with his/her FCC account such as login problems, losing progress, profile page is missing some solutions and so on.
**Beta site** - the issue is related to FCC's beta site.
**bug** - confirmed and reproducible bug in FCC's code and not only (for example, in [Internet Explorer](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/1354)) .
**CHROME** - the issue can be reproduced only in Google Chrome.
**Curriculum Request** - the issue requests a new challenge, adding more explanation to a challenge and so on. Relates to everything that is requested to be added to current curriculum.
**Device Issue** - the issue presents only for a particular device (tablet, smartphone, desktop, etc.).
**Discussing** - the issue is currently on discussion stage.
**duplicate** - the issue is duplicate of another issue and most likely will be closed.
**Easy** - the issue is easy to fix.
**Feature Request** - the issue is a feature request for FCC. For example. ability to [edit or remove](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/1342) bonfires from campers profile page.
**FireFox** - the issue can be reproduced only in Mozilla Firefox.
**help wanted** - owners or issue moderators need campers help to investigate or fix the issue.
**Hikes** - the issue is related to Hikes (they are currently not available in curriculum).
**IE/EDGE** - the issue can be reproduced only in Internet Explorer or Microsoft Edge. Remember that we do **not** support any of these two browsers. FCC officially supports only Google Chrome.
**in progress** - I think this means that currently someone is making a fix for the issue.
**question** - the issue is just a question.
**Resolved** - the issue is resolved/fixed.
**Test Improvement** - the issue should be fixed with a test improvement, which means that someone needs to fix `assert`s.
**todo** - something that needs to be done in future.
**UX** (User eXperience) - in this kind of issues campers share their experience with the issue.
**Wiki** - the issue relates to FCC's GitHub Wiki articles. May be a request to create one. 
Your issue is such a request so I'll add **Wiki** label to this issue.
**wontfix** - the issue is not fixable or won't be fixed in near future. An example of such an issue is [MS Edge browser support](https://github.com/FreeCodeCamp/FreeCodeCamp/issues/1566).

OK, I left some labels not described here: Accessibility, blocked, confirmed, critical-path, enhancement, invalid, QA, reactify, ready and sounding. Hopefully @FreeCodeCamp/owners and other @FreeCodeCamp/issue-moderators will help to finish the list.

**Comment by ahstro** [comment]
I'm by no means an authority on the subject, but here are som of the remaining labels with descriptions that I think are somewhat correct, maybe, possibly.
Please edit them if I'm wrong.

**blocked** - I believe this is for when an issue should not be worked on, due to waiting on the creator to respond to some question or something.
**confirmed** - I think this was the old **QA**. Not sure if it's used anymore.
**citical-path** - IIRC this is used with very obvious or site-breaking bugs
**enhancement** - I think this is sort of like **Feature Request**, possibly more related to improving existing features rather than adding new ones?
**invalid** - Probably an issue with a feature or some code that is no longer on the site?
**QA** - Short for "Quality Assurance". For when a fix has been submitted and it needs to be reviewed before merging.
**reactify** - An issue related to the future reacification of FCC.

**Comment by benschenker** [comment]
**Accessibility** - This is used for issues that don't necessarily break the site but make it harder to use (like when the tests used to evaluate while you typed and would flicker the whole screen).  Also this is relevant to enabling the site to be usable for those who may have vision impairment or disabilities.
**Blocked** - We need more information or the issue is stale.
**Confirmed**  - This is used when the issue that has been reported has been verified (it basically says to the issue creator that they have successfully communicated how to replicate the issue) but no one has started working toward a fix.
**UX** Similar to accessibility but it focuses on an issue that leads to a negative user experience.  If something is more complicated than it needs to be or it is hard to understand how to use a certain feature then this tag is relevant.
**Easy** - This is used to flag issues that aren't critical but should be easy for a camper to fix for their first couple of PRs.  Think of these as the camper gateway to becoming a contributor.

**Comment by AEA2002** [comment]
@bugron, @ahstro, @benschenker,  thank you so much for your replies. They are above and beyond what I was expecting. 

One more question. If an issue has wiki label should a camper write something and submit a pull request or is better to wait for discussion about the topic or a help wanted label?

**Comment by bugron** [comment]
@AEA2002 you are free to open PRs and fix issues. You can create a Wiki article about labels if you want. Don't forget to read [**Guidelines for Contributing**](https://github.com/FreeCodeCamp/FreeCodeCamp/blob/staging/CONTRIBUTING.md) first, thanks.

**Comment by AEA2002** [comment]
@bugron Great! i know all I need to know. Thanks, again.

**Comment by benschenker** [comment]
I think we should keep this open until a wiki entry is created.  This is clearly valuable to have and be up to date.

**Comment by abhisekp** [comment]
Probably, @Rafase282 could create a wiki entry.

**Comment by Rafase282** [comment]
I'm not familiar with the labels and currently without checking (not sure where I can find the info) I would just put the info from here.

Should that not be part of [this](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Creating-a-New-Github-Issue) ?

**Comment by bugron** [comment]
@Rafase282 no, is should be a separate article.

**Comment by BerkeleyTrue** [comment]
Agreed, this is only relevant to issue mods and owners.

I believe this is the only source for label information that exist at the moment.

**Comment by Rafase282** [comment]
Anyone feel free to create the page, no need to ask for permission.

On Mon, Sep 21, 2015, 1:38 AM Berkeley Martinez <notifications@github.com>
wrote:

> Agreed, this is only relevant to issue mods and owners.
>
> I believe this is the only source for label information that exist at the
> moment.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/FreeCodeCamp/FreeCodeCamp/issues/3371#issuecomment-141881165>
> .
>
-- 
Kind Regards,

Rafael Rodríguez
Top freelancer @ UpWork
https://goo.gl/b3vwK8

**Comment by AEA2002** [comment]
> Great! i know all I need to know. Thanks, again.

Thought I knew all I needed to know, but after reading some of the comments I'm not sure. I created a Wiki page on labels and was going save it in the Wiki section. Should I save offline and submit a pull request instead?

**Comment by BerkeleyTrue** [comment]
You can just save directly to the wiki

**Comment by AEA2002** [comment]
@owners, @collaborators

I used the definitions above for a issue labels wiki. However, I do not have descriptions for the following labels:
- Invalid
- Reactify
- Ready
- Sounding
- (label: Critical-Path) What is IIRC?
Thanks

**Comment by BerkeleyTrue** [comment]
* invalid - Not sure whats up with this
* Reactify - Issues related to moving to reactjs
* Sounding - no idea
* Critical Path - This is no longer used

**Comment by bugron** [comment]
@AEA2002 I think IIRC is short for "If I Remember Correctly". @BerkeleyTrue should we remove invalid, Sounding and Critical Path labels if they are not used?

**Comment by BerkeleyTrue** [comment]
I think it is safe to remove them if they are not currently being used.

Do remember that some of these labels (blocked, confirmed, QA) are also tied to Waffle and are automatically added and removed by it (worth mentioning in the article).

**Comment by AEA2002** [comment]
I will remove the unused labels and add info about labels tied to waffle.

IIRC (lol)

**Comment by AEA2002** [comment]
It needs editing (the title especially needs help), but here is the link for  the *label* wiki:
https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Select-Issues-for-Contributing-Using-Labels

**Comment by ahstro** [comment]
While we're on the subject, could someone standardize the capitalization of the labels? Some of them use `Title Case`, other's `all lower case`, and I cringe every time I see FireFox.

**Comment by BerkeleyTrue** [comment]
Besides those tied to Waffle I agree that there should be a standard capitalization. I prefer to keep it all lowercase.

**Comment by Rafase282** [comment]
I feel your pain @BerkeleyTrue

**Comment by ahstro** [comment]
@BerkeleyTrue Can I take the liberty to rename them and possibly add colors to colorless ones and delete unused ones?
+1 to lower case.

**Comment by BerkeleyTrue** [comment]
Yes.

**Comment by ahstro** [comment]
I've lower cased, and in some cases shortened, the names of all labels except `QA` and `Discussing` since I think these are related to the Waffle board and I didn't dare to. Do you know if they're fine to edit as well, @BerkeleyTrue?

I also think we could merge `Feature Request` and `Curriculum Request` into `enhancement`, possibly merge `accessibility` into `ux` or vice versa, and remove the following labels:
* `resolved`
* `question`
* `invalid`
* `ready`
* `performance`
* `critical-path`
* `sounding`
* `todo`

Thoughts, anyone?

**Comment by BerkeleyTrue** [comment]
Do not edit those labels related to Waffle (confirmed, QA, Discussing).

**Comment by ahstro** [comment]
@BerkeleyTrue I didn't, there were any other than `confirmed`, `QA`, `Discussing`, and `blocked`? Any thoughts on the other suggestions?

**Comment by bugron** [comment]
Closing this issue as [the article](https://github.com/FreeCodeCamp/FreeCodeCamp/wiki/Select-Issues-for-Contributing-Using-Labels) is created.
